### PR TITLE
Improve Android finalize release

### PR DESCRIPTION
**Pull request by loremattei**
This PR adds some actions in order to improve the Android release finalization. 
It can be tested in the companion: https://github.com/wordpress-mobile/WordPress-Android/pull/10176

### CODE REVIEW

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_create_xml_release_notes.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  module Actions
+    module SharedValues
+      ANDROID_UPDATE_METADATA_CUSTOM_VALUE = :ANDROID_UPDATE_METADATA_CUSTOM_VALUE
+    end
+
+    class AndroidCreateXmlReleaseNotesAction < Action
+      def self.run(params)
+        require_relative '../../helper/android/android_git_helper.rb'
+
+        relase_notes_path = params[:download_path] + "/release_notes.xml"
```
Comment (jkmassel) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_create_xml_release_notes.rb]: Nitpick… `relase_notes_path` => `release_notes_path`.

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_create_xml_release_notes.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      ANDROID_UPDATE_METADATA_CUSTOM_VALUE = :ANDROID_UPDATE_METADATA_CUSTOM_VALUE
```
Comment (jkmassel) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_create_xml_release_notes.rb]: When running the script I received a warning saying:

```
warning: already initialized constant Fastlane::Actions::SharedValues::ANDROID_UPDATE_METADATA_CUSTOM_VALUE [...] android_create_xml_release_notes.rb:4: warning: previous definition of ANDROID_UPDATE_METADATA_CUSTOM_VALUE was here.

I think this file might be `require`'d more than once, resulting in this warning? I _think_ you could change `=` to `||=` to only initialize it if hasn't been already?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_create_xml_release_notes.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                                        optional: false,
+                                       is_string: true),
+          FastlaneCore::ConfigItem.new(key: :locales,
+                                        env_name: "FL_DOWNLOAD_METADATA_LOCALES",
+                                        description: "The hash with the GLotPress locale and the project locale association",
+                                        is_string: false),
+        ]
+      end
+
+      def self.output
```
Comment (jkmassel) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_create_xml_release_notes.rb]: Should we keep these stubs if we're not using them?

(`output` and `return_value`)

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_hotfix.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        @current_version = Fastlane::Helpers::AndroidVersionHelper.get_release_version()
+        @current_version_alpha = Fastlane::Helpers::AndroidVersionHelper.get_alpha_version()
+        @new_version = Fastlane::Helpers::AndroidVersionHelper.calc_next_hotfix_version(new_version_name, new_version_code)
+        @new_short_version = new_version_name
+        @new_release_branch = "release/#{@new_short_version}"
+      end
+
+      def self.show_config()
+        UI.message("Current version: #{@current_version[Fastlane::Helpers::AndroidVersionHelper::VERSION_NAME]}(#{@current_version[Fastlane::Helpers::AndroidVersionHelper::VERSION_CODE]})")
+        UI.message("New hotifx version: #{@new_version[Fastlane::Helpers::AndroidVersionHelper::VERSION_NAME]}(#{@new_version[Fastlane::Helpers::AndroidVersionHelper::VERSION_CODE]})")
```
Comment (jkmassel) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_hotfix.rb]: Small log typo here (`hotifx` => `hotfix`)

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_final_release.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        @current_version_alpha = Fastlane::Helpers::AndroidVersionHelper.get_alpha_version()
+        @final_version = Fastlane::Helpers::AndroidVersionHelper.calc_final_release_version(@current_version, @current_version_alpha)
+      end
+
+      def self.show_config()
+        vname = Fastlane::Helpers::AndroidVersionHelper::VERSION_NAME
+        vcode = Fastlane::Helpers::AndroidVersionHelper::VERSION_CODE
+        UI.message("Current version: #{@current_version[vname]}(#{@current_version[vcode]})")
+        UI.message("Current alpha version: #{@current_version_alpha[vname]}(#{@current_version_alpha[vcode]})") unless ENV["HAS_ALPHA_VERSION"].nil?
+        UI.message("New Final version: #{@final_version[vname]}(#{@final_version[vcode]})")
```
Comment (jkmassel) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_final_release.rb]: `Final` here (but in this whole file) feels a little odd – could this be `release` instead? Again, fairly nitpicky, but we can only ever have one _truly_ "final" version of the app?

**Comment by loremattei** [comment]
Thanks for the thoughtful review @jkmassel! 🙇

**Pull request merged**